core/optic/shared/src/main/scala/com/useoptic/diff/shapes/JsonLikeTraverser.scala

```diff
@@ -76,7 +76,7 @@ class JsonLikeTraverser(spec: RfcState, visitors: JsonLikeVisitors) {
         })
         val jsonTrail = bodyTrail.withChild(JsonObjectKey(key))
 
-        visitors.objectVisitor.visit(key, value, bodyTrail, resolvedTrail, resolved.bindings)
+        visitors.objectVisitor.visit(key, value, jsonTrail, resolvedTrail, resolved.bindings)
```

Comment:
I think this may be causing some of the misplaced diffs we get in nested objects. each field had the wrong trail :(

Reply:
@devdoshi -- am I missing something? All other behavior seems to work fine with the change